light-client/ibc-client-starknet-types/src/encoding/impls/client_state.rs

```diff
@@ -21,22 +22,27 @@ delegate_components! {
                     symbol!("latest_height"),
                     EncodeLengthDelimitedProtoField<1, UseContext>,
                 >,
+                EncodeField<
+                    symbol!("chain_id"),
+                    EncodeLengthDelimitedProtoField<2, UseContext>,
+                >,
             ]>,
         MutDecoderComponent: DecodeFrom<
             Self,
             CombineEncoders<Product![
                 DecodeRequiredProtoField<1, UseContext>,
+                DecodeRequiredProtoField<2, UseContext>,
             ]>
         >,
     }
 }
 
 impl Transformer for EncodeStarknetClientState {
-    type From = Product![Height];
+    type From = Product![Height, Felt];
 
     type To = StarknetClientState;
 
-    fn transform(product![latest_height]: Self::From) -> Self::To {
-        StarknetClientState { latest_height }
+    fn transform(product![latest_height, chain_id]: Self::From) -> Self::To {
+        StarknetClientState { latest_height, chain_id: chain_id.to_string() }
```

Comment:
This should be the same way we print `ClientId` with `client_type` field with `Felt` type.

https://github.com/informalsystems/ibc-starknet/blob/c13b888f907494910de113845fbbe559d50112bf/relayer/crates/starknet-chain-components/src/types/client_id.rs#L43-L50